### PR TITLE
ED-1691 make functional tests part of CI process

### PR DESCRIPTION
**Pull request by kowalcj0**
With these changes we can run functional tests from CLI using:

```bash
# run both smoke & functional tests
make test

# run both smoke & functional tests in a docker container
make docker_test

# run all functional tests
make functional_tests

# selected scenarios using make
BEHAVE_ARGS="-t certain_tag" make functional_tests

# run functional tests using regular `behave` runner
behave tests/functional/features/

# run selected functional tests using regular `behave` runner
behave tests/functional/features/ -t certain_tag
```

### CODE REVIEW

File: tests/functional/features/environment.py [hunk 1]
```diff
@@ -5,6 +5,10 @@
 from tests.functional.features.context_utils import initialize_scenario_data
 from tests.functional.features.context_utils import patch_context
 from tests.functional.features.utils import init_loggers
+from tests.functional.features.settings import S3_ACCESS_KEY_ID
```
Comment (konradko) [inline tests/functional/features/environment.py]:
```
from tests.functional.features.settings import (
    S3_ACCESS_KEY_ID,
    S3_SECRET_ACCESS_KEY,
    S3_BUCKET,
    S3_REGION
)
```

File: makefile [hunk 2]
```diff
@@ -87,6 +90,14 @@ smoke_tests:
 	$(SET_PYTEST_ENV_VARS); \
 	pytest tests/smoke $(pytest_args)
 
+SET_FUNCTIONAL_ENV_VARS := \
+	export DIRECTORY_API_URL=https://dev.buyer.directory.uktrade.io/api/
```
Comment (konradko) [inline makefile]: this is already set by `SET_PYTEST_ENV_VARS` - should be corrected there

File: tests/functional/README.md [hunk 3]
```diff
@@ -4,7 +4,8 @@ Functional Test for FAB, FAS & SSO
 
 # create virtualenv & install dependencies
 ```bash
-mkvirtualenv .env -p python3.5
```
Comment (konradko) [inline tests/functional/README.md]: `virtualenvwrapper` is better than pure venv :P

File: makefile (hunk 2 above)
Comment (kowalcj0) [inline makefile]: yes it is set, but to a different host, that doesn't expose `api/internal` endpoints

File: tests/functional/README.md (hunk 3 above)
Comment (kowalcj0) [inline tests/functional/README.md]: @konradko I've already changed that, see https://github.com/uktrade/directory-tests/pull/36/commits/16f774558a3309868042366ea07fededc3b898d4

File: tests/functional/features/environment.py (hunk 1 above)
Comment (kowalcj0) [inline tests/functional/features/environment.py]: fixed

File: README.md [hunk 4]
```diff
@@ -19,9 +18,26 @@ Python3
     $ make
 
 ## Running
-Tests can be run either against locally provisioned environment or any other one, as long as ``DIRECTORY_TESTS_DIRECTORY_API_URL``, ``DIRECTORY_TESTS_DIRECTORY_SSO_URL``, ``DIRECTORY_TESTS_DIRECTORY_UI_BUYER_URL`` and ``DIRECTORY_TESTS_DIRECTORY_UI_SUPPLIER_URL`` environment variables are set.
```
Comment (konradko) [inline README.md]: Why is load test readme getting removed?

File: makefile (hunk 2 above)
Comment (konradko) [inline makefile]: it is the same host dev.buyer.directory.uktrade.io is an alias for directory-api-dev.herokuapp.com - please set `DIRECTORY_API_URL` to `https://dev.buyer.directory.uktrade.io`. `/api` should be added in the client init.

File: tests/functional/features/environment.py [hunk 5]
```diff
@@ -7,6 +7,10 @@
     patch_context
 )
 from tests.functional.features.utils import init_loggers
+from tests.functional.features.settings import S3_ACCESS_KEY_ID
```
Comment (konradko) [inline tests/functional/features/environment.py]: should import all in one line

File: README.md [hunk 6]
```diff
@@ -39,6 +55,34 @@ This will run flake8 linting, integration tests and finally load tests. This doe
     make test_load_supplier
     make test_load_sso
 
+3) To run functional tests locally:
```
Comment (konradko) [inline README.md]: should mention running in docker as well

File: README.md (hunk 4 above)
Comment (kowalcj0) [inline README.md]: @konradko It wasn't removed. I've turned it into a list all required env vars. See the changes below.

File: makefile (hunk 2 above)
Comment (kowalcj0) [inline makefile]: Done.

File: README.md (hunk 6 above)
Comment (kowalcj0) [inline README.md]: @konradko ATM there's no need for dedicated `docker_functional_tests` target.
Functional tests are executed as part of `docker_integration_tests` target.

File: tests/functional/features/environment.py (hunk 5 above)
Comment (kowalcj0) [inline tests/functional/features/environment.py]: that's been fixed in a different branch.

File: makefile (hunk 2 above)
Comment (kowalcj0) [inline makefile]: I've restored the correct value of this URL

File: tests/functional/features/environment.py (hunk 5 above)
Comment (kowalcj0) [inline tests/functional/features/environment.py]: I should rather say that it these imports where removed along with redundant assertions in: 16dc7ed790d0acf9bade5df4975fc1a5246bce61

**Pull request merged**